### ISSUE TITLE
Gamepad input support

### ISSUE THREAD

**Issue by nealey** [nealey/vail#9]
I'm embarrassed this didn't occur to me sooner. Lots of people have gamepads, and they're made specifically for timing sensitive button-based input. They even make wireless ones.

**Comment by nealey** [comment]
This demo code I just whipped out should do the trick:

```js
gpButtons = {}
gpPolling = false

function dog(now) {
	let currentButtons = {}
	for (let gp of navigator.getGamepads()) {
		if (gp == null) {
			continue
		}
		currentButtons.key = gp.buttons[0].pressed || gp.buttons[1].pressed
		currentButtons.dit = gp.buttons[2].pressed || gp.buttons[4].pressed
		currentButtons.dah = gp.buttons[3].pressed || gp.buttons[5].pressed
	}
	
	if (currentButtons.key != gpButtons.key) {
		console.log("key", currentButtons.key)
	}
	if (currentButtons.dit != gpButtons.dit) {
		console.log("dit", currentButtons.dit)
	}
	if (currentButtons.dah != gpButtons.dah) {
		console.log("dah", currentButtons.dah)
	}
	gpButtons = currentButtons
	
	requestAnimationFrame(dog)
}
			
function gpc(event) {
	// Polling could be computationally expensive,
	// especially on devices with a power budget, like phones.
	// To be considerate, we only start polling if a gamepad appears.
	if (! gpPolling) {
		gpPolling = true
		dog(event.timeStamp)
	}
}
window.addEventListener("gamepadconnected", gpc)
```

**Comment by nealey** [comment]
This code will fail miserably if there are two gamepads connected. Need to keep an array of "last state" objects, one per gamepad. Logical or states of all pads to determine global state, and then compare to previous global state to become edge-triggered.

**Comment by nealey** [comment]
Implemented with 8e6e18f41386183a5df5c12824b4a07c8f4c8cdb

**Comment by nealey** [comment]
Oops, no, 94914dc is unrelated to this
